### PR TITLE
Operator.sol full coverage

### PR DESCRIPTION
**Pull request by samigabor**
Well, almost full coverage... _msgData() not coverer

### CODE REVIEW

File: packages/network-contracts/contracts/OperatorTokenomics/Operator.sol [hunk 1]
```diff
@@ -185,8 +186,10 @@ contract Operator is Initializable, ERC2771ContextUpgradeable, IERC677Receiver,
 
         ERC20Upgradeable.__ERC20_init(operatorTokenName, operatorTokenName);
 
-        // DEFAULT_ADMIN_ROLE is needed (by factory) for setting modules
+        // DEFAULT_ADMIN_ROLE is needed (by factory) for setting modules and policies
         _setupRole(DEFAULT_ADMIN_ROLE, _msgSender());
+        // DEFAULT_ADMIN_ROLE is needed (by the contract owner) for setting policies
+        _setupRole(DEFAULT_ADMIN_ROLE, ownerAddress);
```
Comment (jtakalai) [inline packages/network-contracts/contracts/OperatorTokenomics/Operator.sol:192]:
```suggestion
```

File: packages/network-contracts/test/hardhat/OperatorTokenomics/Operator.test.ts [hunk 2]
```diff
@@ -182,6 +313,27 @@ describe("Operator contract", (): void => {
...
+            await (await token.connect(delegator).approve(operator.address, parseEther("1000"))).wait()
+            await expect(token.connect(delegator).transferAndCall(operator.address, parseEther("1000"), data))
+                .to.emit(operator, "Delegated").withArgs(delegator.address, parseEther("1000"))
+        })
+    
+        it("allows delegate without delegation policy being set", async function(): Promise<void> {
+            const { token } = sharedContracts
+            await setTokens(delegator, "1000")
+            const operator = await deployOperator(operatorWallet)
+            await (await operator.connect(operatorWallet).setDelegationPolicy(hardhatEthers.constants.AddressZero, 0)).wait()
```
Comment (jtakalai) [inline packages/network-contracts/test/hardhat/OperatorTokenomics/Operator.test.ts:332]:
```suggestion
            const operator = await deployOperatorContract(operatorWallet, {
                delegationPolicyAddress: hardhatEthers.constants.AddressZero
            })
```
and add overrides to deployOperator and deployOperatorContract

**Review by jtakalai — CHANGES_REQUESTED** [review]

File: packages/network-contracts/contracts/OperatorTokenomics/Operator.sol (hunk 1 above)
Comment (samigabor) [reply packages/network-contracts/contracts/OperatorTokenomics/Operator.sol]: Removed

File: packages/network-contracts/test/hardhat/OperatorTokenomics/Operator.test.ts (hunk 2 above)
Comment (samigabor) [reply packages/network-contracts/test/hardhat/OperatorTokenomics/Operator.test.ts]: Updated

File: packages/network-contracts/test/hardhat/OperatorTokenomics/deployTestContracts.ts [hunk 3]
```diff
@@ -24,6 +24,8 @@ export type TestContracts = {
     defaultDelegationPolicy: IDelegationPolicy;
     defaultExchangeRatePolicy: IExchangeRatePolicy;
     defaultUndelegationPolicy: IUndelegationPolicy;
+    testExchangeRatePolicy: IExchangeRatePolicy;
+    testExchangeRatePolicy2: IExchangeRatePolicy;
```
[the remark this answers is not in the document]
Comment (samigabor) [reply packages/network-contracts/test/hardhat/OperatorTokenomics/deployTestContracts.ts]: Updated

File: packages/network-contracts/test/hardhat/OperatorTokenomics/deployTestContracts.ts [hunk 4]
```diff
@@ -109,6 +118,10 @@ export async function deployTestContracts(signer: Wallet): Promise<TestContracts
     const defaultExchangeRatePolicy = await (await getContractFactory("DefaultExchangeRatePolicy", { signer })).deploy()
     const defaultUndelegationPolicy = await (await getContractFactory("DefaultUndelegationPolicy", { signer })).deploy()
 
+    // test policies
+    const testExchangeRatePolicy = await (await getContractFactory("TestExchangeRatePolicy", { signer })).deploy()
+    const testExchangeRatePolicy2 = await (await getContractFactory("TestExchangeRatePolicy2", { signer })).deploy()
+
```
[the remark this answers is not in the document]
Comment (samigabor) [reply packages/network-contracts/test/hardhat/OperatorTokenomics/deployTestContracts.ts]: Updated

**Review by jtakalai — APPROVED** [review]

**Pull request merged**